cloud_functions/utils.js

```diff
@@ -19,23 +19,45 @@ const coerceToError = e => {
   }
 }
 
-// TODO: Share this code with the bard GAE app (will probably require a more sophisticated build/deploy process).
-// However, this implementation is slightly different in that it lets exceptions escape. This is
-// appropriate for how it's used in the context of this Cloud Function, but might not be appropriate
-// for use in the GAE app or may require changes to the app. Letting exceptions escape instead of
-// returning a magic null value is a simpler, better expression of the behavior. That makes it worth
-// diverging the code now. Future de-duplication efforts should consider using this simpler
-// implementation in both places.
+// TODO: Share the code below with the bard GAE app (will probably require a more sophisticated build/deploy process).
+
+class Response {
+  constructor(status, data) {
+    this.status = status
+    this.data = data
+  }
+}
+
+const promiseHandler = fn => (req, res) => {
+  const handleValue = value => {
+    if (value instanceof Response) {
+      res.status(value.status).send(value.data)
+    } else {
+      console.error(value)
+      res.status(500).send(value.toString())
```

Comment:
Me: _this is not java, this can't cause an NPE, this is not java..._

Reply:
I mean, it could if you throw or return `undefined` or `null`, but it's clear by inspection AND a basic test that we don't do that.